src/sif/truckcost.cc

```diff
@@ -535,6 +542,8 @@ Cost TruckCost::TransitionCost(const baldr::DirectedEdge* edge,
   uint32_t idx = pred.opp_local_idx();
   Cost c = base_transition_cost(node, edge, &pred, idx);
   c.secs = OSRMCarTurnDuration(edge, node, idx);
+  c.cost +=
+      destination_only_penalty_ * (edge->destonly_hgv() && !pred.destonly() && !edge->destonly());
```

Comment:
this is good but wont you'll still get penalized from the base transition cost right if the regular dest only is set

Reply:
you can over in base cost just not apply the regular one if you are hgv

Reply:
handled it in `base_transition_cost`